### PR TITLE
JSON API documentation improvements

### PR DESCRIPTION
**Pull request by leo-da**
Closes: #3732

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/json-api/index.rst [hunk 1]
```diff
@@ -12,22 +12,22 @@ tracker
...
-API </app-dev/index>` to access *basic active contract set
-functionality*:
+The **JSON API** provides a significantly simpler way than :doc:`the Ledger
+API </app-dev/index>` to interact with a ledger by providing *basic active contract set functionality*:
 
-- creating contracts,
-- exercising choices on contracts, and
-- querying the current active contract set.
+- create contract,
+- fetch active contract by contract ID or contract key,
```
Comment (S11001001) [inline docs/source/json-api/index.rst]: Keep the description here informal, describing categories of functionality not enumerating features; fetch is already covered by "query the current active contract set".

File: docs/source/json-api/index.rst [hunk 2]
```diff
@@ -152,243 +152,143 @@ add the subprotocols ``jwt.token.copy-paste-token-here`` and
...
-Request
--------
+- 200 - OK
+- 400 - Bad Request (Client Error)
+- 401 - Unauthorized, authentication required
+- 404 - Not Found
+- 500 - Internal Server Error
 
-application/json body, formatted according to the :doc:`search-query-language`:
+If client's HTTP request reaches an API endpoint, the corresponding response will always contain a JSON object with ``status`` field and either ``errors`` or ``result``:
```
Comment (S11001001) [inline docs/source/json-api/index.rst]:
```suggestion
If client's HTTP GET or POST request reaches an API endpoint, the corresponding response will always contain a JSON object with ``status`` field and either ``errors`` or ``result``:
```

File: docs/source/json-api/index.rst [hunk 3]
```diff
@@ -569,26 +549,29 @@ Contract Found Response
         }
     }
 
-Lookup by Contract Key
-----------------------
+Fetch Contract by Contract Key
```
Comment (hurryabit) [inline docs/source/json-api/index.rst]:
```suggestion
Fetch Contract by Key
```

File: docs/source/json-api/index.rst [hunk 4]
```diff
@@ -626,3 +612,261 @@ Contract Found Response
             "templateId": "11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Account:Account"
         }
     }
+
+
+Contract Search, All Templates
+==============================
+
+List all currently active contracts for all known templates. Note that the retrieved contracts do not get persisted into query store database.
```
Comment (hurryabit) [inline docs/source/json-api/index.rst]: Is there any explanation of what the "query store database" is? I couldn't find any.

File: docs/source/json-api/index.rst [hunk 5]
```diff
@@ -152,243 +152,143 @@ add the subprotocols ``jwt.token.copy-paste-token-here`` and
...
-Response
---------
+Where:
+
+- ``templateId`` is the contract template identifier, which can be formatted as either:
+
+  + ``"<package ID>:<module>:<entity>"`` or
+  + ``"<module>:<entity>"`` if contract template can be uniquely identified by it's module and entity name.
+
+- ``argument`` field contains contract fields as defined in the DAML template and formatted according to :doc:`lf-value-specification`.
```
[the remark this answers is not in the document]
Comment (leo-da) [reply docs/source/json-api/index.rst]: @hurryabit it is renamed only in the response. See #3826 :
> I suggest we rename the argument field of the contracts you receive from the JSON API to payload.

This particular JSON is a create command, a request sent to JSON API.

[the remark this answers is not in the document]
Comment (leo-da) [reply docs/source/json-api/index.rst]: if you want to change it to the `payload` in the requests, this should be addressed in a different ticket.

**Pull request merged**